### ISSUE TITLE
Could not resolve Rebus.Routing.IRouter with decorator depth 0 - registrations: Rebus.Injection.Injectionist+Handler

### ISSUE THREAD

**Issue by larsey** [rebus-org/Rebus#772]
Hello, I have this problem that I don't understand. I sat Rebus up in a sandbox and made a test with a simple class with just a string property and it was working great. Then I copied all the steps (or did I?) and implemented the same receiver code as below. Now I get an exception that Configure throws. It builds and runs (then crashes and burns)

**Exception**:
Could not resolve Rebus.Routing.IRouter with decorator depth 0 - registrations: Rebus.Injection.Injectionist+Handler

**Includes**:
using Rebus.Activation;
using Rebus.Config;
using Rebus.Handlers;
using Rebus.Logging;
using Rebus.Routing.TypeBased;

**Code**:
using (var activator = new BuiltinHandlerActivator())
{
    activator.Register(() => new Handler());

    try
    {
        Configure.With(activator)
            .Logging(l => l.ColoredConsole(minLevel: LogLevel.Info))
            .Transport(t => t.UseMsmq($"foo-queue"))
            .Routing(r => r.TypeBased().MapAssemblyOf<SimpleDataPoint>("otherfoo-queue"))
            .Start();
    }
    catch(Exception ex)
    {

    }

    activator.Bus.Subscribe<SimpleDataPoint>().Wait();

    Console.WriteLine("Running forever, press [ENTER] to quit.");
    Console.ReadLine();
}

**Stacktrace**:
   ved Rebus.Injection.Injectionist.ResolutionContext.Get[TService]()
   ved Rebus.Config.RebusConfigurer.<>c__DisplayClass12_0.<Start>b__25(IResolutionContext c)
   ved Rebus.Injection.Injectionist.Resolver`1.InvokeResolver(IResolutionContext context)
   ved Rebus.Injection.Injectionist.ResolutionContext.Get[TService]()
   ved Rebus.Config.RebusConfigurer.<>c__DisplayClass12_0.<Start>b__26(IResolutionContext c)
   ved Rebus.Injection.Injectionist.Resolver`1.InvokeResolver(IResolutionContext context)
   ved Rebus.Injection.Injectionist.ResolutionContext.Get[TService]()
   ved Rebus.Injection.Injectionist.Get[TService]()
   ved Rebus.Config.RebusConfigurer.Start()

**Comment by mookid8000** [comment]
Could you add a
```csharp
Console.WriteLine(ex);
```
in the `catch` block?

Your stacktrace does not seem to contain the full exception info.

**Comment by larsey** [comment]
Hello mookid,

I did. It complained it was missing Microsoft.Azure.Storage library. I added it to the subscriber application and it runs.

My setup is that I have a shared class library that is being used by the subscriber application. The subscriber is not using anything else then the SimpleDataPoint class from this library. I'm unsure why I'm getting this error. Why is Rebus throwing this, is it mapping the whole library?

I guess this can be closed.
